pkgs/servers/nosql/dragonflydb/default.nix

```diff
@@ -0,0 +1,110 @@
+{ fetchFromGitHub
+, fetchurl
+, lib
+, stdenv
+, double-conversion
+, gperftools
+, mimalloc
+, rapidjson
+, liburing
+, xxHash
+, abseil-cpp_202111
+, gbenchmark
+, glog
+, gtest
+, jemalloc
+, gcc-unwrapped
+, autoconf
+, autoconf-archive
+, automake
+, cmake
+, ninja
+, boost
+, libunwind
+, libtool
+, openssl
+}:
+
+let
+  pname = "dragonflydb";
+  version = "0.1.0";
+
+  src = fetchFromGitHub {
+    owner = pname;
+    repo = "dragonfly";
+    rev = "v${version}";
+    hash = "sha256-P6WMW/n+VezWDXGagT4B+ZYyCp8oufDV6MTrpKpLZcs=";
+    fetchSubmodules = true;
+  };
+
+  # Needed exactly 5.4.4 for patch to work
+  lua = fetchurl {
+    url = "https://github.com/lua/lua/archive/refs/tags/v5.4.4.tar.gz";
+    hash = "sha256-L/ibvqIqfIuRDWsAb1ukVZ7c9GiiVTfO35mI7ZD2tFA=";
+  };
+in
+stdenv.mkDerivation {
+
+  inherit pname version src;
+
+  postPatch = ''
+    mkdir -p ./build/{third_party,_deps}
+    cp -R --no-preserve=ownership ${double-conversion.src} ./build/third_party/dconv
+    cp -R --no-preserve=ownership ${gperftools.src} ./build/third_party/gperf
+    cp -R --no-preserve=ownership ${mimalloc.src} ./build/third_party/mimalloc
+    cp -R --no-preserve=ownership ${rapidjson.src} ./build/third_party/rapidjson
+    cp -R --no-preserve=ownership ${liburing.src} ./build/third_party/uring
+    cp -R --no-preserve=ownership ${xxHash.src} ./build/third_party/xxhash
+    cp -R --no-preserve=ownership ${abseil-cpp_202111.src} ./build/_deps/abseil_cpp-src
+    cp -R --no-preserve=ownership ${gbenchmark.src} ./build/_deps/benchmark-src
+    cp -R --no-preserve=ownership ${glog.src} ./build/_deps/glog-src
+    cp -R --no-preserve=ownership ${gtest.src} ./build/_deps/gtest-src
+    chmod -R u+w ./build/{third_party,_deps}
```

Comment:
Can't we copy them without ownership and mode? Maybe we need to fix the perms for ``./build/third_part`` first.

Reply:
Some files like (`configure`) have the execute bit set and hence the build fails if the mode is removed. I'll see if I can patch this somehow.

Reply:
Done. Had to chmod u+x the `configure` file and now it works.